pcsx2/USB/Win32/Config_usb.cpp

```diff
@@ -0,0 +1,243 @@
+/*  PCSX2 - PS2 Emulator for PCs
+ *  Copyright (C) 2002-2020  PCSX2 Dev Team
+ *
+ *  PCSX2 is free software: you can redistribute it and/or modify it under the terms
+ *  of the GNU Lesser General Public License as published by the Free Software Found-
+ *  ation, either version 3 of the License, or (at your option) any later version.
+ *
+ *  PCSX2 is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ *  without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR
+ *  PURPOSE.  See the GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License along with PCSX2.
+ *  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "PrecompiledHeader.h"
+#include "AppCoreThread.h"
+#include "../USB.h"
+#include "resource_usb.h"
+#include "Config_usb.h"
+#include "../deviceproxy.h"
+#include "../usb-pad/padproxy.h"
+#include "../usb-mic/audiodeviceproxy.h"
+#include "../configuration.h"
+#include "../shared/inifile_usb.h"
+
+HINSTANCE hInstUSB;
+extern bool configChanged;
+
+void SysMessageA(const char* fmt, ...)
+{
+	va_list list;
+	char tmp[512];
+
+	va_start(list, fmt);
+	vsprintf_s(tmp, 512, fmt, list);
+	va_end(list);
+	MessageBoxA(0, tmp, "USB Msg", 0);
+}
+
+void SysMessageW(const wchar_t* fmt, ...)
+{
+	va_list list;
+	wchar_t tmp[512];
+
+	va_start(list, fmt);
+	vswprintf_s(tmp, 512, fmt, list);
+	va_end(list);
+	MessageBoxW(0, tmp, L"USB Msg", 0);
+}
+
+void SelChangedAPI(HWND hW, int port)
+{
+	int sel = SendDlgItemMessage(hW, port ? IDC_COMBO_API1_USB : IDC_COMBO_API2_USB, CB_GETCURSEL, 0, 0);
+	int devtype = SendDlgItemMessage(hW, port ? IDC_COMBO1_USB : IDC_COMBO2_USB, CB_GETCURSEL, 0, 0);
+	if (devtype == 0)
+		return;
+	devtype--;
+	auto& rd = RegisterDevice::instance();
+	auto devName = rd.Name(devtype);
+	auto apis = rd.Device(devtype)->ListAPIs();
+	auto it = apis.begin();
+	std::advance(it, sel);
+	changedAPIs[std::make_pair(port, devName)] = *it;
+}
+
+void PopulateAPIs(HWND hW, int port)
+{
+	SendDlgItemMessage(hW, port ? IDC_COMBO_API1_USB : IDC_COMBO_API2_USB, CB_RESETCONTENT, 0, 0);
+	int devtype = SendDlgItemMessage(hW, port ? IDC_COMBO1_USB : IDC_COMBO2_USB, CB_GETCURSEL, 0, 0);
+	if (devtype == 0)
+		return;
+	devtype--;
+	auto& rd = RegisterDevice::instance();
+	auto dev = rd.Device(devtype);
+	auto devName = rd.Name(devtype);
+	auto apis = dev->ListAPIs();
+
+	std::string selApi = GetSelectedAPI(std::make_pair(port, devName));
+
+	std::string var;
+	if (!LoadSetting(nullptr, port, rd.Name(devtype), N_DEVICE_API, str_to_wstr(var)))
```

Comment:
`str_to_wstr` ... doesn't it go poof in the ether?
Might as well add back `LoadSetting(...., std::string& v)` for windows for now if there are some other plans to change settings system than to sprinkel random `std::wstring tmp;`s around?

Reply:
Yeah definitely a solid idea, I just tried to keep the changes as minimal as possible for the initial merge but it seems like this is out of the window, guess I'll try to make it a bit cleaner.

Reply:
Rewriting everything using std::string requires rewriting the entire ini loading mechanism, so instead I'll wait for the config rewrite to take care of that and fix issues with the current system in the meanwhile.

Reply:
Uh? It's just https://github.com/jackun/USBqemu-wheel/blob/bd1d68208810621cc8c3ca812b1fd39d43e83fcb/src/configuration.cpp#L93 , no? :D

Reply:
and inifile which uses wstring :)

Reply:
just `wchar_t *` it :P

Reply:
then why move to std::string at all if it's not to do a complete switch and still have wstring/std::string conversions everywhere?

Reply:
Oh, if you want to do so then yeah. I was thinking if anyone tries currently to run it on windows, they're gonna have a bad day.

Reply:
Ah yeah I've double checked the LoadSetting that are borked, there's two, I already fixed some of them earlier. This should fix all the windows config issues afaik, and i'll postpone the new config design to the yaml PR, good for you?

Reply:
ok